string dataset_dir
---
bool success
